era/base.js

```diff
@@ -398,70 +453,93 @@ export function handler_castCanMissNoCoefficient(threatValue) {
   return (ev, fight) => {
     let t = ev.type;
     if (t === "cast") {
-      threatFunctions.sourceThreatenTarget(ev, fight, threatValue, false);
+      threatFunctions.sourceThreatenTarget({
+        ev,
+        fight,
+        amount: threatValue,
+        useThreatCoeffs: false,
+      });
     } else if (t === "damage") {
-      threatFunctions.sourceThreatenTarget(ev, fight, -threatValue, false);
+      threatFunctions.sourceThreatenTarget({
+        ev,
+        fight,
+        amount: -threatValue,
+        useThreatCoeffs: false,
+      });
     }
   };
 }
 
 export function handler_modDamage(multiplier) {
   return (ev, fight) => {
     if (ev.type !== "damage") return;
-    threatFunctions.sourceThreatenTarget(
+    threatFunctions.sourceThreatenTarget({
       ev,
       fight,
-      ev.amount + (ev.absorbed || 0),
-      true,
-      multiplier
-    );
+      amount: ev.amount + (ev.absorbed || 0),
+      multiplier,
+      debugLabel: `modDamage`,
+    });
   };
 }
 export function handler_modHeal(multiplier) {
   return (ev, fight) => {
     if (ev.type !== "heal") return;
-    threatFunctions.unitThreatenEnemiesSplit(
+    threatFunctions.unitThreatenEnemiesSplit({
       ev,
-      "source",
+      unit: "source",
       fight,
-      (multiplier * ev.amount) / 2
-    );
+      amount: ev.amount,
+      multiplier: multiplier * 0.5,
+      debugLabel: `modHeal (${multiplier} * 0.5)`,
+    });
   };
 }
 
 export function handler_modDamagePlusThreat(multiplier, bonus) {
   return (ev, fight) => {
     if (ev.type !== "damage" || ev.hitType > 6 || ev.hitType === 0) return;
-    threatFunctions.sourceThreatenTarget(
+    threatFunctions.sourceThreatenTarget({
       ev,
       fight,
-      multiplier * (ev.amount + (ev.absorbed || 0)) + bonus
-    );
+      amount: ev.amount + (ev.absorbed || 0),
+      multiplier,
+      bonusThreat: bonus,
+      debugLabel: `modDamagePlusThreat (dmg * ${multiplier}) + ${bonus}`,
+    });
   };
 }
 
 export function handler_damage(ev, fight) {
   if (ev.type !== "damage") return;
-  threatFunctions.sourceThreatenTarget(
+  threatFunctions.sourceThreatenTarget({
     ev,
     fight,
-    ev.amount + (ev.absorbed || 0)
-  );
+    amount: ev.amount + (ev.absorbed || 0),
+  });
 }
 
 export function handler_heal(ev, fight) {
   if (ev.type !== "heal") return;
-  threatFunctions.unitThreatenEnemiesSplit(ev, "source", fight, ev.amount / 2);
+  threatFunctions.unitThreatenEnemiesSplit({
+    ev,
+    unit: "source",
+    fight,
+    amount: ev.amount,
+    multiplier: 0.5,
```

Comment:
Most other changes in this file are extracting out the amount and multiplier so that we can have better debug info.